### PR TITLE
ENH: Table sorting and ds004229 updates

### PR DESCRIPTION
**Pull request by larsoner**
- [x] Changelog has been updated (`docs/source/changes.md`)
- [x] Update config for ds004229 v1.0.3
- [x] Add [table sorting](https://squidfunk.github.io/mkdocs-material/reference/data-tables/#sortable-tables) so if you want to find all examples that for example use ICA you can sort by the ICA column and sort it by checked/unchecked.

Someday I dream of having a more compact table layout and sticky first row/col but after some time trying several CSS solutions I was unable to do so!

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -65,6 +65,10 @@ copyright: Copyright &copy; MNE-BIDS-Pipeline authors
 
 extra_css:
     - css/extra.css
+# https://squidfunk.github.io/mkdocs-material/reference/data-tables/
+extra_javascript:
+  - https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js
```
Comment (hoechenberger) [inline docs/mkdocs.yml:70]: We should vendor this file instead of pulling it from the CDN

**Comment by hoechenberger** [comment]
> Someday I dream of having a more compact table layout and sticky first row/col but after some time trying several CSS solutions I was unable to do so!

@larsoner The tables we use for metadata are quite nice IMHO

https://output.circle-artifacts.com/output/job/ae9d8f27-6bde-4b79-b1ba-6f020efbc2ea/artifacts/0/reports/ERP_CORE/sub-015_ses-MMN_task-MMN_report.html#Epochs__before_cleaning-Metadata

File: docs/mkdocs.yml (hunk 1 above)
Comment (hoechenberger) [reply docs/mkdocs.yml:70]: Then again, I think we're not vendoring our emojis either, so …

Comment (larsoner) [reply docs/mkdocs.yml:70]: Let's wait to vendor until later, then. Maybe in the end we don't need to?

**Comment by larsoner** [comment]
Nicely sortable now

https://output.circle-artifacts.com/output/job/cea69984-fa63-49ea-b485-fc15c9c90724/artifacts/0/site/examples/examples.html#demonstrated-features

**Comment by hoechenberger** [comment]
Nice! But I do t get any visual indicator of this functionality until I actually start clicking on the header cells. So this feature is not easily discoverable 🤔

**Review by agramfort — APPROVED** [review]

**Comment by hoechenberger** [comment]
> Nice! But I don't get any visual indicator of this functionality until I actually start clicking on the header cells. So this feature is not easily discoverable 🤔

Ah, it happens on hover! Fair enough

**Pull request merged**